src/App.tsx

```diff
@@ -40,14 +40,17 @@ const StyledPageBody = styled.div`
 `;
 
 const App: React.FC = () => {
-  if (!USE_MOCK_DATA) {
-    Amplify.configure(awsConfig);
-  }
-
   const dispatch = useDispatch();
   const { t } = useTranslation('feedback');
   const { closeSnackbar, enqueueSnackbar } = useSnackbar();
   const feedback = useSelector((store: RootStore) => store.feedback);
+  const auth = useSelector((store: RootStore) => store.auth);
+
+  useEffect(() => {
+    if (!USE_MOCK_DATA && !auth.isLoggedIn) {
```

Comment:
Kunne denne useEffecten vært vært merget med den på linje 68?
Ser ut til at hver gang denne kjører vil også else-blokken til siste useEffecten også kjøre

Reply:
kan gjøre det, men tenkte at det kunne være greit å skille på configen og å hente brukerdata. den dependencyen auth.isLoggedIn på den andre useEffecten er feil ser jeg nå. den skal egentlig ikke være der

Reply:
Hva er best?

a)
`  useEffect(() => {
    if (!USE_MOCK_DATA && !auth.isLoggedIn) {
      Amplify.configure(awsConfig);
    }
  }, [auth.isLoggedIn]);`
`  useEffect(() => {
    if (USE_MOCK_DATA) {
      dispatch(setUser(mockUser));
    } else {
      dispatch(getCurrentAuthenticatedUser());
      Hub.listen('auth', data => hubListener(data, dispatch));
      return () => Hub.remove('auth', data => hubListener(data, dispatch));
    }
  }, [dispatch]);`

eller b)
`  useEffect(() => {
    if (USE_MOCK_DATA) {
      dispatch(setUser(mockUser));
    } else {
      if (!auth.isLoggedIn) {
        Amplify.configure(awsConfig);
      }
      dispatch(getCurrentAuthenticatedUser());
      Hub.listen('auth', data => hubListener(data, dispatch));
      return () => Hub.remove('auth', data => hubListener(data, dispatch));
    }
  }, [dispatch, auth.isLoggedIn]);`

Reply:
b

Reply:
hvis ikke begge har `auth.isLoggedIn` som dependency er det nok like greit å ha det som to useEffects ja